bin/dummy_cuckoo_api.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+""" Dummy Cuckoo API
+
+to be used while developing PeekabooAV
+Allows testing the API link to Cuckoo Sandbox
+
+# requires
+# pip install flask flask-restful """
+
+import json
+import random
+import sys
+import flask
+from flask_restful import abort, Resource, Api
+
+
+def run():
+    """ Run the API """
+
+    app = flask.Flask(__name__)
+    app.debug = True
+    cuckoo_api = Api(app)
+
+    queue = Queue()
+
+    cuckoo_api.add_resource(Status, '/cuckoo/status', resource_class_args=[queue])
+    cuckoo_api.add_resource(View, '/tasks/view/<job_id>', resource_class_args=[queue])
+    cuckoo_api.add_resource(Report, '/tasks/report/<job_id>', resource_class_args=[queue])
+    cuckoo_api.add_resource(Create, '/tasks/create/file', resource_class_args=[queue])
+
+    app.run(port='5002')
+
+
+class Queue():
+    """ Internal data structure to keep track of submitted tasks """
+    def __init__(self):
+        self.nextid = 1
+        self.jobs = []
+
+    def register_job(self):
+        """ Adds a new job to the Queue and returns its id """
+        self.nextid = self.nextid + 1
+        self.jobs.append(self.nextid - 1)
+        print("New job created %d. Queue %s" % (self.nextid-1, self.jobs))
+        return self.nextid - 1
+
+    def delete_job(self, job_id):
+        """ Removes a job from the Queue """
+        self.jobs.remove(job_id)
+        print("Remove job {}. Queue %s".format(job_id) % self.jobs)
+
+
+class QueueResource(Resource):
+    """ Intermediate class to extend normal Resource by Queue """
+    def __init__(self, queue):
+        self.queue = queue
+
+
+class Status(QueueResource):
+    """ Resource to get Dummy Cuckoo's status on tasks """
+    def get(self):
+        """ Returns number of tasks in the respective state """
+        result = {'tasks':
```

Comment:
pycodestyle says:
```
bin/dummy_cuckoo_api.py:64:23: E127 continuation line over-indented for visual indent
bin/dummy_cuckoo_api.py:65:80: E501 line too long (83 > 79 characters)
bin/dummy_cuckoo_api.py:70:18: E124 closing bracket does not match visual indentation
```
That should be easily fixed:
```
@@ -60,14 +60,14 @@ class Status(QueueResource):
     """ Resource to get Dummy Cuckoo's status on tasks """
     def get(self):
         """ Returns number of tasks in the respective state """
-        result = {'tasks':
-                      {
-                          'reported': self.queue.nextid - 1 - len(self.queue.jobs),
-                          'running': len(self.queue.jobs),
-                          'total': self.queue.nextid - 1,
-                          'dummyAPIqueue': self.queue.jobs
-                      }
-                 }
+        result = {
+            'tasks': {
+                'reported': self.queue.nextid - 1 - len(self.queue.jobs),
+                'running': len(self.queue.jobs),
+                'total': self.queue.nextid - 1,
+                'dummyAPIqueue': self.queue.jobs,
+            }
+        }
         return flask.jsonify(result)

```
Also note the comma on `dummyAPIqueue` for whenever we extend or reorder that list.

Reply:
Addressed in new commits.